libsolidity/ast/Types.h

```diff
@@ -869,6 +888,8 @@ class ArraySliceType: public ReferenceType
 
 protected:
 	std::vector<std::tuple<std::string, TypePointer>> makeStackItems() const override;
+	std::vector<Type const*> decomposition() const override { return {&m_arrayType}; }
```

Comment:
Shouldn't it be `m_arrayType->decomposition()` instead? I don't think a slice is composed of full arrays :) At least counting it as such does not make sense in size calculations.

Reply:
Sounds right :).

Reply:
Done, https://github.com/ethereum/solidity/pull/9818/files#diff-c903efe2e4a71d75dc2eed4252cb24d8R897